src/or/connection.c

```diff
@@ -2604,25 +2621,37 @@ connection_read_proxy_handshake(connection_t *conn)
   return ret;
 }
 
+struct replacement_s
+{
+  connection_t *old_conn;
+  port_cfg_t *new_port;
+};
+
 /** Given a list of listener connections in <b>old_conns</b>, and list of
  * port_cfg_t entries in <b>ports</b>, open a new listener for every port in
  * <b>ports</b> that does not already have a listener in <b>old_conns</b>.
  *
  * Remove from <b>old_conns</b> every connection that has a corresponding
  * entry in <b>ports</b>.  Add to <b>new_conns</b> new every connection we
- * launch.
+ * launch. If we may need to perform socket rebind when creating new
+ * listener that replaces old one, create a <b>replacement_s</b> struct
+ * for affected pair  and add it to <b>replacements</b>. For more
+ * information, see ticket #17873.
```

Comment:
We try not to use this style: the documentation for the code should stand by itself, and not require the bugtracker for help understanding why it is that way.